.github/workflows/protocol_tests.yml

```diff
@@ -94,8 +96,13 @@ jobs:
 
       - name: Run Everything just in case something was missed
         # can't use gas limit because some setUp function use more than the limit
-        run: forge test -vvv #TODO this should ignore integration tests
-
+        # Excludes integration tests (`RegistryIntegrationTest`), and other tests that are
+        # excluded globally in `foundry.toml` like `RandomTest` and `BLS12_381Passthrough`.
+        # Unfortunately, when you specify a `--no-match-contract` flag, it overrides any global
+        # exclusions specified in `foundry.toml`. Therefore, somewhat annoyingly, the exclusions here
+        # have to cover all cases, namely the integration tests and the global exclusions.
+        run: forge test -vvv --no-match-contract "RegistryIntegrationTest|RandomTest|BLS12_381Passthrough"
```

Comment:
Any reason to not add `RegistryIntegrationTest` in the `foundry.toml` instead of using a flag here? 

It shouldn't be a problem when running the integration test, since `--match-path test-sol/integration/Integration.t.sol` will overwrite the `foundry.toml`

Reply:
> `--match-path test-sol/integration/Integration.t.sol` will overwrite the `foundry.toml`

Great idea, I didn't consider this option. That would probably be cleaner 👍

Reply:
Unfortunately, I couldn't find a cleaner way to solve this, and still satisfy the testing needs. I tried various combinations of `foundry.toml` and `forge test` flags. 

The requirements are:

1. [`workflows/protocol_tests.yml`](https://github.com/celo-org/celo-monorepo/blob/9c913ca0e88c490813786b58850d2ddc08316e13/.github/workflows/protocol_tests.yml#L97-L98)

    Run everything
    BUT not integration tests
    BUT not globally excluded tests

    ```sh
    forge test \
    -vvv \
    --no-match-contract RegistryIntegrationTest
    ```

1. [`migrations_sol/run_integration_tests_in_anvil.sh`](https://github.com/celo-org/celo-monorepo/blob/9c913ca0e88c490813786b58850d2ddc08316e13/packages/protocol/migrations_sol/run_integration_tests_in_anvil.sh#L9-L10)

    Run ONLY integration tests

    ```sh
    forge test \
    -vvv \
    --match-contract RegistryIntegrationTest \
    --fork-url http://127.0.0.1:$ANVIL_PORT
    ```

1. [`foundry.toml`](https://github.com/celo-org/celo-monorepo/blob/9c913ca0e88c490813786b58850d2ddc08316e13/packages/protocol/foundry.toml#L22)

    NEVER RUN tests in `BLS12Passthrough.sol`
    NEVER RUN tests in `RandomTest.sol`
    
    ```toml
    # `Random.sol` is excluded, but I'm not sure why. It was already excluded so I'm leaving it here.
    no_match_contract = "RandomTest"
    
    no_match_test = "skip"
    
    # `BLS12Passthrough.sol` is excluded, because it's tested in the celo-blockain repo as 
    # described here: https://github.com/celo-org/celo-monorepo/pull/10240
    no_match_path = "contracts/common/libraries/test/BLS12Passthrough.sol"
    ```

Reply:
This is refactored and done in https://github.com/celo-org/celo-monorepo/pull/11002/commits/255f467731edfcd12ef8de492946c66c59724d2f